Dubai Real Estate Market Intelligence Platform

In [1]:
import pandas as pd 

In [2]:
df1=pd.read_csv("C:\\Dubai Real Estate Market Intelligence Platform\\data\\raw\\transactions_2026-07-16_17-30-13_0001.csv")
df2=pd.read_csv("C:\\Dubai Real Estate Market Intelligence Platform\\data\\raw\\transactions_2026-07-16_17-30-13_0002.csv")

In [3]:
#Merging 2 csv files into one using pandas concat method since both files have same columns
df = pd.concat([df1, df2], ignore_index=True)

In [5]:
df.shape

(1748953, 47)

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1748953 entries, 0 to 1748952
Data columns (total 47 columns):
 #   Column                Dtype  
---  ------                -----  
 0   actual_worth          float64
 1   area_id               int64  
 2   area_name_ar          object 
 3   area_name_en          object 
 4   building_name_ar      object 
 5   building_name_en      object 
 6   has_parking           int64  
 7   instance_date         object 
 8   master_project_ar     object 
 9   master_project_en     object 
 10  meter_rent_price      float64
 11  meter_sale_price      float64
 12  nearest_landmark_ar   object 
 13  nearest_landmark_en   object 
 14  nearest_mall_ar       object 
 15  nearest_mall_en       object 
 16  nearest_metro_ar      object 
 17  nearest_metro_en      object 
 18  no_of_parties_role_1  float64
 19  no_of_parties_role_2  float64
 20  no_of_parties_role_3  float64
 21  procedure_area        float64
 22  procedure_id          int64  
 23  procedu

In [9]:
#Droping all arabic columns from the dataframe since they are not needed for analysis
ar_cols = [c for c in df.columns if c.endswith('_ar')]
print(ar_cols)#Checking the arabic columns to be dropped

['area_name_ar', 'building_name_ar', 'master_project_ar', 'nearest_landmark_ar', 'nearest_mall_ar', 'nearest_metro_ar', 'procedure_name_ar', 'project_name_ar', 'property_sub_type_ar', 'property_type_ar', 'property_usage_ar', 'reg_type_ar', 'rooms_ar', 'trans_group_ar']


In [10]:
df = df.drop(columns=ar_cols)

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1748953 entries, 0 to 1748952
Data columns (total 33 columns):
 #   Column                Dtype  
---  ------                -----  
 0   actual_worth          float64
 1   area_id               int64  
 2   area_name_en          object 
 3   building_name_en      object 
 4   has_parking           int64  
 5   instance_date         object 
 6   master_project_en     object 
 7   meter_rent_price      float64
 8   meter_sale_price      float64
 9   nearest_landmark_en   object 
 10  nearest_mall_en       object 
 11  nearest_metro_en      object 
 12  no_of_parties_role_1  float64
 13  no_of_parties_role_2  float64
 14  no_of_parties_role_3  float64
 15  procedure_area        float64
 16  procedure_id          int64  
 17  procedure_name_en     object 
 18  project_name_en       object 
 19  project_number        float64
 20  property_sub_type_en  object 
 21  property_sub_type_id  float64
 22  property_type_en      object 
 23  propert

In [12]:
# Strip the _en suffix since Arabic columns are gone (no ambiguity now)
df.columns = [c[:-3] if c.endswith('_en') else c for c in df.columns]

# Standard cleaning: lowercase, strip whitespace, replace spaces with underscores
df.columns = (
    df.columns
    .str.strip()
    .str.lower()
    .str.replace(' ', '_')
    .str.replace(r'[^\w]', '', regex=True)  # remove any other special chars
)

print(df.columns.tolist())

['actual_worth', 'area_id', 'area_name', 'building_name', 'has_parking', 'instance_date', 'master_project', 'meter_rent_price', 'meter_sale_price', 'nearest_landmark', 'nearest_mall', 'nearest_metro', 'no_of_parties_role_1', 'no_of_parties_role_2', 'no_of_parties_role_3', 'procedure_area', 'procedure_id', 'procedure_name', 'project_name', 'project_number', 'property_sub_type', 'property_sub_type_id', 'property_type', 'property_type_id', 'property_usage', 'reg_type', 'reg_type_id', 'rent_value', 'rooms', 'transaction_id', 'trans_group', 'trans_group_id', 'load_timestamp']


In [13]:
df.head()

,actual_worth,area_id,area_name,building_name,has_parking,instance_date,master_project,meter_rent_price,meter_sale_price,nearest_landmark,...,property_type_id,property_usage,reg_type,reg_type_id,rent_value,rooms,transaction_id,trans_group,trans_group_id,load_timestamp
0,181912.0,445,Jabal Ali First,J & G BUILDING 2,1,2013-06-25,DMCC Master Community,NaN,3701.16,Sports City Swimming Academy,...,3,Industrial,Existing Properties,1,NaN,NaN,1-11-2013-18300,Sales,1,2026-07-16T17:27:47.000Z
1,3124888.0,447,Al Khairan First,Palace Residences Creek Blue Tower 1,1,2024-08-12,Dubai Creek Harbour,NaN,28157.22,NaN,...,3,Residential,Off-Plan Properties,0,NaN,2 B/R,1-102-2024-59929,Sales,1,2026-07-16T17:27:47.000Z
2,1450000.0,333,Madinat Dubai Almelaheyah,THE ANWA BY OMNIYAT,1,2026-05-14,Dubai Maritime City,NaN,11497.90,Burj Khalifa,...,3,Residential,Existing Properties,1,NaN,2 B/R,2-13-2026-13386,Mortgages,2,2026-07-16T17:27:47.000Z
3,297000.0,484,Nadd Hessa,CRYSTALS,1,2025-03-24,Silicon Oasis,NaN,7115.48,IMG World Adventures,...,3,Residential,Existing Properties,1,NaN,Studio,2-39-2025-249,Mortgages,2,2026-07-16T17:27:47.000Z
4,2623000.0,350,Al Thanyah Fifth,NaN,0,2013-09-25,Jumeirah Park,NaN,4806.58,Sports City Swimming Academy,...,4,Commercial,Existing Properties,1,NaN,NaN,1-110-2013-2045,Sales,1,2026-07-16T17:27:47.000Z


In [14]:
print(df['instance_date'].head(10))
print(df['instance_date'].dtype)

0    2013-06-25
1    2024-08-12
2    2026-05-14
3    2025-03-24
4    2013-09-25
5    2024-12-16
6    2025-11-19
7    2025-12-29
8    2023-11-29
9    2020-02-05
Name: instance_date, dtype: object
object


In [17]:
df['instance_date'] = pd.to_datetime(df['instance_date'], format='%Y-%m-%d', errors='coerce')

df['year'] = df['instance_date'].dt.year
df['month'] = df['instance_date'].dt.month

In [18]:
print(df['instance_date'].isna().sum())

4


In [ ]:
df = df.dropna(subset=['instance_date'])#dropped rows with invalid dates negligible because they are only 3 rows out of 1.5 million rows

In [20]:
df.head()

,actual_worth,area_id,area_name,building_name,has_parking,instance_date,master_project,meter_rent_price,meter_sale_price,nearest_landmark,...,reg_type,reg_type_id,rent_value,rooms,transaction_id,trans_group,trans_group_id,load_timestamp,year,month
0,181912.0,445,Jabal Ali First,J & G BUILDING 2,1,2013-06-25,DMCC Master Community,NaN,3701.16,Sports City Swimming Academy,...,Existing Properties,1,NaN,NaN,1-11-2013-18300,Sales,1,2026-07-16T17:27:47.000Z,2013.0,6.0
1,3124888.0,447,Al Khairan First,Palace Residences Creek Blue Tower 1,1,2024-08-12,Dubai Creek Harbour,NaN,28157.22,NaN,...,Off-Plan Properties,0,NaN,2 B/R,1-102-2024-59929,Sales,1,2026-07-16T17:27:47.000Z,2024.0,8.0
2,1450000.0,333,Madinat Dubai Almelaheyah,THE ANWA BY OMNIYAT,1,2026-05-14,Dubai Maritime City,NaN,11497.90,Burj Khalifa,...,Existing Properties,1,NaN,2 B/R,2-13-2026-13386,Mortgages,2,2026-07-16T17:27:47.000Z,2026.0,5.0
3,297000.0,484,Nadd Hessa,CRYSTALS,1,2025-03-24,Silicon Oasis,NaN,7115.48,IMG World Adventures,...,Existing Properties,1,NaN,Studio,2-39-2025-249,Mortgages,2,2026-07-16T17:27:47.000Z,2025.0,3.0
4,2623000.0,350,Al Thanyah Fifth,NaN,0,2013-09-25,Jumeirah Park,NaN,4806.58,Sports City Swimming Academy,...,Existing Properties,1,NaN,NaN,1-110-2013-2045,Sales,1,2026-07-16T17:27:47.000Z,2013.0,9.0


In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1748949 entries, 0 to 1748952
Data columns (total 35 columns):
 #   Column                Dtype         
---  ------                -----         
 0   actual_worth          float64       
 1   area_id               int64         
 2   area_name             object        
 3   building_name         object        
 4   has_parking           int64         
 5   instance_date         datetime64[ns]
 6   master_project        object        
 7   meter_rent_price      float64       
 8   meter_sale_price      float64       
 9   nearest_landmark      object        
 10  nearest_mall          object        
 11  nearest_metro         object        
 12  no_of_parties_role_1  float64       
 13  no_of_parties_role_2  float64       
 14  no_of_parties_role_3  float64       
 15  procedure_area        float64       
 16  procedure_id          int64         
 17  procedure_name        object        
 18  project_name          object        
 19  proje

In [22]:
df['year'] = df['year'].astype(int)
df['month'] = df['month'].astype(int)

In [23]:
df[['year', 'month']].dtypes

year     int32
month    int32
dtype: object

In [24]:
df.head()

,actual_worth,area_id,area_name,building_name,has_parking,instance_date,master_project,meter_rent_price,meter_sale_price,nearest_landmark,...,reg_type,reg_type_id,rent_value,rooms,transaction_id,trans_group,trans_group_id,load_timestamp,year,month
0,181912.0,445,Jabal Ali First,J & G BUILDING 2,1,2013-06-25,DMCC Master Community,NaN,3701.16,Sports City Swimming Academy,...,Existing Properties,1,NaN,NaN,1-11-2013-18300,Sales,1,2026-07-16T17:27:47.000Z,2013,6
1,3124888.0,447,Al Khairan First,Palace Residences Creek Blue Tower 1,1,2024-08-12,Dubai Creek Harbour,NaN,28157.22,NaN,...,Off-Plan Properties,0,NaN,2 B/R,1-102-2024-59929,Sales,1,2026-07-16T17:27:47.000Z,2024,8
2,1450000.0,333,Madinat Dubai Almelaheyah,THE ANWA BY OMNIYAT,1,2026-05-14,Dubai Maritime City,NaN,11497.90,Burj Khalifa,...,Existing Properties,1,NaN,2 B/R,2-13-2026-13386,Mortgages,2,2026-07-16T17:27:47.000Z,2026,5
3,297000.0,484,Nadd Hessa,CRYSTALS,1,2025-03-24,Silicon Oasis,NaN,7115.48,IMG World Adventures,...,Existing Properties,1,NaN,Studio,2-39-2025-249,Mortgages,2,2026-07-16T17:27:47.000Z,2025,3
4,2623000.0,350,Al Thanyah Fifth,NaN,0,2013-09-25,Jumeirah Park,NaN,4806.58,Sports City Swimming Academy,...,Existing Properties,1,NaN,NaN,1-110-2013-2045,Sales,1,2026-07-16T17:27:47.000Z,2013,9


In [ ]:
df.isnull().sum()[df.isnull().sum() > 0]#Checking for null values in each column

building_name            496745
master_project           235413
meter_rent_price        1712126
nearest_landmark         355275
nearest_mall             557554
nearest_metro            546833
no_of_parties_role_1        971
no_of_parties_role_2        971
no_of_parties_role_3        971
project_name             465876
project_number           465876
property_sub_type        344704
property_sub_type_id     344704
rent_value              1712126
rooms                    370203
dtype: int64

In [ ]:
df = df.drop(columns=['meter_rent_price', 'rent_value'])#dropped columns with null values since they are not needed for analysis

In [ ]:
df.duplicated().sum() # no duplicates found in the dataset

0

In [29]:
df.head()

,actual_worth,area_id,area_name,building_name,has_parking,instance_date,master_project,meter_sale_price,nearest_landmark,nearest_mall,...,property_usage,reg_type,reg_type_id,rooms,transaction_id,trans_group,trans_group_id,load_timestamp,year,month
0,181912.0,445,Jabal Ali First,J & G BUILDING 2,1,2013-06-25,DMCC Master Community,3701.16,Sports City Swimming Academy,Marina Mall,...,Industrial,Existing Properties,1,NaN,1-11-2013-18300,Sales,1,2026-07-16T17:27:47.000Z,2013,6
1,3124888.0,447,Al Khairan First,Palace Residences Creek Blue Tower 1,1,2024-08-12,Dubai Creek Harbour,28157.22,NaN,NaN,...,Residential,Off-Plan Properties,0,2 B/R,1-102-2024-59929,Sales,1,2026-07-16T17:27:47.000Z,2024,8
2,1450000.0,333,Madinat Dubai Almelaheyah,THE ANWA BY OMNIYAT,1,2026-05-14,Dubai Maritime City,11497.90,Burj Khalifa,Dubai Mall,...,Residential,Existing Properties,1,2 B/R,2-13-2026-13386,Mortgages,2,2026-07-16T17:27:47.000Z,2026,5
3,297000.0,484,Nadd Hessa,CRYSTALS,1,2025-03-24,Silicon Oasis,7115.48,IMG World Adventures,NaN,...,Residential,Existing Properties,1,Studio,2-39-2025-249,Mortgages,2,2026-07-16T17:27:47.000Z,2025,3
4,2623000.0,350,Al Thanyah Fifth,NaN,0,2013-09-25,Jumeirah Park,4806.58,Sports City Swimming Academy,Ibn-e-Battuta Mall,...,Commercial,Existing Properties,1,NaN,1-110-2013-2045,Sales,1,2026-07-16T17:27:47.000Z,2013,9


In [32]:
#drop column load_timestamp since it is not needed for analysis
df = df.drop(columns=['load_timestamp'])

In [33]:
df.head()

,actual_worth,area_id,area_name,building_name,has_parking,instance_date,master_project,meter_sale_price,nearest_landmark,nearest_mall,...,property_type_id,property_usage,reg_type,reg_type_id,rooms,transaction_id,trans_group,trans_group_id,year,month
0,181912.0,445,Jabal Ali First,J & G BUILDING 2,1,2013-06-25,DMCC Master Community,3701.16,Sports City Swimming Academy,Marina Mall,...,3,Industrial,Existing Properties,1,NaN,1-11-2013-18300,Sales,1,2013,6
1,3124888.0,447,Al Khairan First,Palace Residences Creek Blue Tower 1,1,2024-08-12,Dubai Creek Harbour,28157.22,NaN,NaN,...,3,Residential,Off-Plan Properties,0,2 B/R,1-102-2024-59929,Sales,1,2024,8
2,1450000.0,333,Madinat Dubai Almelaheyah,THE ANWA BY OMNIYAT,1,2026-05-14,Dubai Maritime City,11497.90,Burj Khalifa,Dubai Mall,...,3,Residential,Existing Properties,1,2 B/R,2-13-2026-13386,Mortgages,2,2026,5
3,297000.0,484,Nadd Hessa,CRYSTALS,1,2025-03-24,Silicon Oasis,7115.48,IMG World Adventures,NaN,...,3,Residential,Existing Properties,1,Studio,2-39-2025-249,Mortgages,2,2025,3
4,2623000.0,350,Al Thanyah Fifth,NaN,0,2013-09-25,Jumeirah Park,4806.58,Sports City Swimming Academy,Ibn-e-Battuta Mall,...,4,Commercial,Existing Properties,1,NaN,1-110-2013-2045,Sales,1,2013,9


In [34]:
 #Fill remaining nulls
for col in ['building_name', 'master_project', 'nearest_landmark', 'nearest_mall', 'nearest_metro', 'project_name']:
    df[col] = df[col].fillna('Unknown')

df['rooms'] = df['rooms'].fillna('N/A')
df['property_sub_type'] = df['property_sub_type'].fillna('Unknown')

In [35]:
 #Check for zero/negative values in money & size columns — these would break growth and price calcs:
print((df['actual_worth'] <= 0).sum())
print((df['meter_sale_price'] <= 0).sum())
print((df['procedure_area'] <= 0).sum())

0
8
0


In [37]:
#Check trans_group values analysis is Sales-focused, but the dataset likely also has Mortgages, Gifts, etc
print(df['trans_group'].value_counts())

trans_group
Sales        1339750
Mortgages     344037
Gifts          65162
Name: count, dtype: int64


In [38]:
#Check area_name and property_type for messy/inconsistent labels
print(df['area_name'].nunique())
print(df['property_type'].value_counts())

258
property_type
Unit        1252204
Villa        305007
Land         155245
Building      36493
Name: count, dtype: int64


In [39]:
#Outlier check on actual_worth — real estate data often has extreme outliers (e.g. AED 1 placeholder transactions, or huge portfolio deals) that will skew averages:
df['actual_worth'].describe()

count    1.748949e+06
mean     3.878681e+06
std      6.036761e+07
min      1.000000e+00
25%      7.500000e+05
50%      1.357199e+06
75%      2.520000e+06
max      1.378694e+10
Name: actual_worth, dtype: float64

In [40]:
# drop meter_sale_price <= 0 and procedure_area <= 0 since they are invalid values for analysis
df = df[df['meter_sale_price'] > 0]

In [41]:
print(df['property_type'].nunique())
print(df['property_type'].unique())

4
['Unit' 'Villa' 'Land' 'Building']


In [42]:
# See how many are suspiciously low (e.g. under 10,000 AED — likely non-market transactions like gifts between family)
print((df['actual_worth'] < 10000).sum())

# See how many are extreme highs (top 0.1%)
print(df['actual_worth'].quantile([0.99, 0.999, 0.9999]))

387
0.9900    3.620652e+07
0.9990    2.400000e+08
0.9999    1.424247e+09
Name: actual_worth, dtype: float64


## Handling Outliers in `actual_worth`

Initial inspection of `actual_worth` revealed a heavily right-skewed distribution:

- **Minimum**: 1 AED
- **Median**: ~1.36M AED
- **Mean**: ~3.88M AED (far above median, indicating skew)
- **Maximum**: ~13.8B AED

This skew is driven by two issues:

1. **Near-zero values** (387 rows under 10,000 AED) — likely gifts, family transfers, or data entry errors rather than genuine market sales.
2. **Extreme high-value outliers** — the 99th percentile sits at ~36M AED, but the 99.99th percentile reaches ~1.4B AED, suggesting a small number of large portfolio or land deals.

Left uncorrected, these outliers would distort YoY growth rates, area-level averages, and the Area Ranking model — a handful of mega-deals could make an entire area appear as a "growth star" in a single year.

### Approach

```python
# Drop clearly non-market transactions (near-zero value)
df = df[df['actual_worth'] >= 10000]

# Cap extreme outliers at the 99th percentile for analysis purposes
# (raw column is preserved; capped version used for growth/ranking calcs)
cap = df['actual_worth'].quantile(0.99)
df['actual_worth_capped'] = df['actual_worth'].clip(upper=cap)
```

- Rows below 10,000 AED are removed outright, as they don't represent real market activity.
- Values above the 99th percentile are **capped, not removed** — this preserves row counts (useful for transaction volume metrics) while preventing extreme values from distorting averages and growth calculations.
- The original `actual_worth` column is retained for metrics like **Total AED Value**, where the true (uncapped) sum matters.
- The new `actual_worth_capped` column should be used for **average price**, **YoY growth %**, and the **Area Ranking model**, where outlier sensitivity is a concern.

**Note:** Median (not mean) is still recommended for area-level price comparisons, since real estate prices are naturally skewed even after capping.

In [43]:
# Drop clearly non-market transactions (near-zero value)
df = df[df['actual_worth'] >= 10000]

# Cap extreme outliers at 99th percentile for analysis purposes
# (keep the raw column, add a capped version for growth/ranking calcs)
cap = df['actual_worth'].quantile(0.99)
df['actual_worth_capped'] = df['actual_worth'].clip(upper=cap)

In [44]:
df.head()

,actual_worth,area_id,area_name,building_name,has_parking,instance_date,master_project,meter_sale_price,nearest_landmark,nearest_mall,...,property_usage,reg_type,reg_type_id,rooms,transaction_id,trans_group,trans_group_id,year,month,actual_worth_capped
0,181912.0,445,Jabal Ali First,J & G BUILDING 2,1,2013-06-25,DMCC Master Community,3701.16,Sports City Swimming Academy,Marina Mall,...,Industrial,Existing Properties,1,N/A,1-11-2013-18300,Sales,1,2013,6,181912.0
1,3124888.0,447,Al Khairan First,Palace Residences Creek Blue Tower 1,1,2024-08-12,Dubai Creek Harbour,28157.22,Unknown,Unknown,...,Residential,Off-Plan Properties,0,2 B/R,1-102-2024-59929,Sales,1,2024,8,3124888.0
2,1450000.0,333,Madinat Dubai Almelaheyah,THE ANWA BY OMNIYAT,1,2026-05-14,Dubai Maritime City,11497.90,Burj Khalifa,Dubai Mall,...,Residential,Existing Properties,1,2 B/R,2-13-2026-13386,Mortgages,2,2026,5,1450000.0
3,297000.0,484,Nadd Hessa,CRYSTALS,1,2025-03-24,Silicon Oasis,7115.48,IMG World Adventures,Unknown,...,Residential,Existing Properties,1,Studio,2-39-2025-249,Mortgages,2,2025,3,297000.0
4,2623000.0,350,Al Thanyah Fifth,Unknown,0,2013-09-25,Jumeirah Park,4806.58,Sports City Swimming Academy,Ibn-e-Battuta Mall,...,Commercial,Existing Properties,1,N/A,1-110-2013-2045,Sales,1,2013,9,2623000.0


In [46]:
df.info(memory_usage='deep')

<class 'pandas.core.frame.DataFrame'>
Index: 1748554 entries, 0 to 1748952
Data columns (total 33 columns):
 #   Column                Dtype         
---  ------                -----         
 0   actual_worth          float64       
 1   area_id               int64         
 2   area_name             object        
 3   building_name         object        
 4   has_parking           int64         
 5   instance_date         datetime64[ns]
 6   master_project        object        
 7   meter_sale_price      float64       
 8   nearest_landmark      object        
 9   nearest_mall          object        
 10  nearest_metro         object        
 11  no_of_parties_role_1  float64       
 12  no_of_parties_role_2  float64       
 13  no_of_parties_role_3  float64       
 14  procedure_area        float64       
 15  procedure_id          int64         
 16  procedure_name        object        
 17  project_name          object        
 18  project_number        float64       
 19  prope

# Data Cleaning Summary — Dubai Real Estate Market Intelligence Platform

## Overview
Raw transaction data sourced from Dubai Land Department (via Dubai Pulse) was cleaned and prepared for market growth analysis, area ranking, and property segment comparison.

**Starting shape:** 1,748,953 rows × 47 columns
**Final shape:** 1,748,554 rows × 33 columns

---

## 1. Column Reduction
- Dropped all Arabic-language duplicate columns (`_ar` suffix), retaining only English (`_en`) versions.
- Renamed columns by stripping the `_en` suffix and standardizing to lowercase snake_case.
- Dropped `meter_rent_price` and `rent_value` — both ~98% null, as the dataset is predominantly sales transactions rather than rentals.

## 2. Date Handling
- Converted `instance_date` from string to `datetime64` using `pd.to_datetime(..., format='%Y-%m-%d', errors='coerce')`.
- 4 rows contained corrupted/out-of-range date values (e.g. `1422-11-23`) and were dropped after conversion.
- Extracted `year` and `month` as integer columns for time-based analysis.

## 3. Duplicate Check
- Checked both full-row duplicates and `transaction_id` duplicates.
- **Result:** zero duplicates found — no action needed.

## 4. Null Handling
| Column(s) | Action |
|---|---|
| `meter_rent_price`, `rent_value` | Dropped (98% null) |
| `building_name`, `master_project`, `nearest_landmark`, `nearest_mall`, `nearest_metro`, `project_name` | Filled with `'Unknown'` |
| `rooms` | Filled with `'N/A'` (mostly null for non-residential property types) |
| `property_sub_type` | Filled with `'Unknown'` |
| `no_of_parties_role_1/2/3` | Negligible nulls (971 rows, 0.06%) — left as-is |

## 5. Invalid Value Removal
- Dropped 8 rows where `meter_sale_price <= 0`.
- Dropped 387 rows where `actual_worth < 10,000 AED` — identified as non-market transactions (e.g. gifts, family transfers) rather than genuine sales.

## 6. Outlier Treatment
`actual_worth` showed severe right-skew:
- Mean (3.88M AED) far exceeded median (1.36M AED)
- 99.99th percentile reached ~1.4B AED, driven by a small number of large portfolio/land deals

**Approach:** rather than removing outliers outright, the original `actual_worth` was preserved for metrics requiring true totals (e.g. Total AED Value), while a new column `actual_worth_capped` was created by clipping values at the 99th percentile for use in growth rates, averages, and the Area Ranking model.

```python
cap = df['actual_worth'].quantile(0.99)
df['actual_worth_capped'] = df['actual_worth'].clip(upper=cap)
```

Skewness was confirmed visually via histogram and boxplot comparisons of raw vs. capped values.

## 7. Final Validation
- Confirmed `property_type` is clean with only 4 categories: `Unit`, `Villa`, `Land`, `Building`.
- Confirmed `trans_group` contains three legitimate categories: `Sales`, `Mortgages`, `Gifts` — analysis will filter to `Sales` for market growth and area ranking.
- Verified final dtypes: correct `datetime64`, `int`, `float`, and `object` types across all 33 columns.

---

## Notes for Analysis Phase
- Use **`actual_worth_capped`** and **median** (not mean) for area-level price comparisons to avoid outlier distortion.
- Filter to **`trans_group = 'Sales'`** for core market growth and area ranking metrics; `Mortgages` and `Gifts` can support a secondary transaction-mix analysis.
- Dataset is now ready to load into SQLite/Postgres for SQL-based analysis.